### PR TITLE
Blocks: Fix More/Next dash line

### LINKED ISSUE #10817: “More” and “Page break” blocks: dashed line missing in the editor.
**Describe the bug**
The dashed line that should appear behind the “More” and “Page break” blocks is missing.

**To Reproduce**
1. Add a “More” (or “Page break”) block.
2. Inspect the block's CSS and see that there is a dashed line border specified in the `::before` pseudo element that it's not appearing on the editor. 

**Screenshots**

screen-recording-2018-10-19-17-04-44

**Expected behavior**
The dashed line should be displayed to maintain visual compatibility with the `more` tag, used on the Classic Editor, as seen below:

more-tag-classic-editor

**Solution**
This can be solved simply by defining the z-index of the parent element (`.editor-block-list__block-edit`), as seen on the animated GIF above.

**Desktop:**
 - MacOS X 10.12
 - Chrome 69

**Additional context**
 - Gutenberg 4.0.0
 - WordPress 4.9.8
 - Twenty Seventeen.
 - No other plugins.

### PR DESCRIPTION
**Pull request by aduth**
Closes #10817

This pull request seeks to resolve an issue where the horizontal line intended to be shown with the More / Next blocks.

Before|After
---|---
before|after

**Testing instructions:**

Verify a dashed line(s) is shown adjacent the input of the More / Next block.

Fixes #10817.

### CODE REVIEW

**Comment by ocean90** [comment]
Is this a fix for #10817 / alternative for #10882?

**Comment by aduth** [comment]
@ocean90 Good catch, it is. I had this one sitting around for a while and threw it up without checking for more recent issues.

To #10882, cc @brentswisher :

- I personally prefer the approach here as opposed to continuing a `z-index` battle, since we can rely on the order of the DOM and use `position: relative` to surface the input above the line.
- The input length issue is technically separate, one also considered in #8751, and one which isn't technically fixed there nor in #10822 by certain characters. It's one I spent a considerable amount trying to resolve through an `AutosizeInput` component, but apparently measuring the width of text consistently across browsers is fraught with issues, and from which I separated this minimal change out from.

File: packages/block-library/src/nextpage/editor.scss [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 	// Label
 	> span {
+		position: relative;
```
Comment (brentswisher) [inline packages/block-library/src/nextpage/editor.scss]:
```suggestion
```
This doesn't need to be added, it exists for this block already two lines below, just the more block was missing it

**Comment by brentswisher** [comment]
@aduth I am fine with the `position:relative` fix, even if the z-index was the preferred method, `assets/stylesheets/_z-index.scss` would be the place for it. I'm still mostly stumbling around in the dark here ;) I did add a note that the position: relative already existed on the next block so your new line isn't needed. 

I'll close my pull request and take a look at #8751 if I want to follow up with that issue some more. 

And if you don't mind me asking, regarding an issue like that with two small mostly unrelated bugs, what would the proper etiquette have been?  Two separate pull requests that both referenced the original issue?

File: packages/block-library/src/nextpage/editor.scss (hunk 1 above)
Comment (aduth) [reply packages/block-library/src/nextpage/editor.scss]: >This doesn't need to be added

Ah, good eye! Thanks.

**Comment by aduth** [comment]
>And if you don't mind me asking, regarding an issue like that with two small mostly unrelated bugs, what would the proper etiquette have been? Two separate pull requests that both referenced the original issue?

In a perfect world, there's one pull request for one issue, where one issue prescribes at most one task. In reality, you can expect pull requests covering multiple issues, or issues containing multiple tasks that require multiple pull requests to close. 

>I noticed a weird quirk in Chrome that we may or may not want to fix. The text is ever so slightly cropped:

The width measuring is part of what I'd started down the path of addressing here, but ended up being complex enough that I'd wanted to defer it for it's own pull request. See also #8750.

**Review by youknowriad — APPROVED** [review]
LGTM 👍 

Granted it doesn't fix all the calculations, but it resolves one part of the issue.

**Pull request merged**